ledger-core/virtual/execute/execute.go

```diff
@@ -387,14 +378,96 @@ func (s *SMExecute) stepDeduplicateUsingPendingsTable(ctx smachine.ExecutionCont
-		// TODO: do we have result? if we have result then we should resend result
+	if isDuplicate && isActive {
 		ctx.Log().Warn("duplicate found as pending request")
 		return ctx.Stop()
 	}
 
-	if ctx.AcquireForThisStep(pendingListFilled).IsNotPassed() {
-		return ctx.Sleep().ThenRepeat()
+	return ctx.Jump(s.stepDeduplicateThroughPreviousExecutor)
+}
+
+type DeduplicationBargeInKey struct {
+	LookAt   pulse.Number // TODO fill it correctly
+	Callee   reference.Global
+	Outgoing reference.Global
+}
+
+func (s *SMExecute) stepDeduplicateThroughPreviousExecutor(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	msg := payload.VFindCallRequest{
+		LookAt:   s.Payload.CallOutgoing.GetPulseNumber(),
+		Callee:   s.execution.Object,
+		Outgoing: s.execution.Outgoing,
+	}
+
+	bargeInCallback := ctx.NewBargeInWithParam(func(param interface{}) smachine.BargeInCallbackFunc {
+		res, ok := param.(*payload.VFindCallResponse)
+		if !ok || res == nil {
+			panic(throw.IllegalValue())
+		}
+
+		return func(ctx smachine.BargeInContext) smachine.StateUpdate {
+			s.VFindCallResponse = res
+			return ctx.WakeUp()
+		}
+	})
+
+	bargeInKey := DeduplicationBargeInKey{
+		Callee:   msg.Callee,
+		Outgoing: msg.Outgoing,
+	}
+
+	if !ctx.PublishGlobalAliasAndBargeIn(bargeInKey, bargeInCallback) {
+		return ctx.Error(throw.E("failed to publish bargeInCallback"))
+	}
+
+	s.messageSender.PrepareAsync(ctx, func(goCtx context.Context, svc messagesender.Service) smachine.AsyncResultFunc {
+		err := svc.SendRole(goCtx, &msg, node.DynamicRoleVirtualExecutor, s.execution.Object, s.pulseSlot.PulseData().PrevPulseNumber())
+		return func(ctx smachine.AsyncResultContext) {
+			if err != nil {
+				ctx.Log().Error("failed to send message", err)
+			}
+		}
+	}).WithoutAutoWakeUp().Start()
+
+	return ctx.Sleep().ThenJump(s.stepWaitFindCallResponse)
+}
+
+func (s *SMExecute) stepWaitFindCallResponse(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	if s.VFindCallResponse == nil {
+		ctx.Sleep().ThenRepeat()
+	}
+	return ctx.Jump(s.stepProcessFindCallResponse)
+}
+
+func (s *SMExecute) stepProcessFindCallResponse(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	switch s.VFindCallResponse.Status {
+	case payload.FoundCall:
+		if s.VFindCallResponse.CallResult == nil {
+			ctx.Log().Warn("request found on previous executor, but there was no result")
+			return ctx.Stop()
+		}
+
+		ctx.Log().Warn("request found on previous executor, resend result")
+
+		target := s.Meta.Sender
+		s.messageSender.PrepareAsync(ctx, func(goCtx context.Context, svc messagesender.Service) smachine.AsyncResultFunc {
+			err := svc.SendTarget(goCtx, s.VFindCallResponse.CallResult, target)
+			return func(ctx smachine.AsyncResultContext) {
+				if err != nil {
+					ctx.Log().Error("failed to send message", err)
+				}
+			}
+		}).WithoutAutoWakeUp().Start()
+
+		return ctx.Jump(s.stepFinishRequest)
+
+	case payload.MissingCall:
+	case payload.UnknownCall:
+		// execute it
+		break
```

Comment:
go has break by default. This break is not required

Reply:
done